File: programme/insertion-sort/insertionSort.js
// Write a program for insertion sort in JavaScript
function insertionSort(arr){
    let n = arr.length;
    let i,j,key;
	for (i = 1; i < n; i++){
		key = arr[i];
		j = i - 1;
		while (j >= 0 && arr[j] > key){
			arr[j + 1] = arr[j];
			j = j - 1;
		}
		arr[j + 1] = key;
	}
    return arr;
}


let arr = [14, 33, 27, 35 ,10];
console.log("Input: ", arr)
console.log("Output: ", insertionSort(arr));

File: programme/insertion-sort/insertion-sort.py
# Write a programme for insertion sort

def insertion_sort(arr):

    for i in range(1, len(arr)):

        curr = arr[i]

        j = i-1

        while j >=0 and curr < arr[j] :
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = curr

    return arr
 
arr = [5, 4, 1, 6, 2]

print("Unsorted list, ", arr)
print("Sorted list,   ", insertion_sort(arr))